### ISSUE TITLE
Vendor list changes (BE)

### ISSUE THREAD

**Issue by MariusJulius** [keeleinstituut/tv-tolkevarav#631]
@MariusJulius @thenouan 

1. About vendor price-list sorting. Currently BE has implemented the price endpoint to accept only one sort by and sort_order value. BE changes are needed to make it like a Wiki _It has to be possible to use several filtering button at once (e.g if filtering is done by language and we want to filter the selection by tags or roles)_

2. 100 per_page aslo BE problem. BE should make changes to accept 100 per_page.
error massage: per_page: ["The per page field must not be greater than 50."]
`curl 'https://api.dev.tolkevarav.eki.ee/translation-order/api/prices?per_page=100&page=1' \
  -H 'authority: api.dev.tolkevarav.eki.ee' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJBRzRYOXR2SVhubkoxWUhiYnJHZmdYdnNYNEVVU2xGalk0RkZSekY5MjlRIn0.eyJleHAiOjE3MDE0MjM2MTksImlhdCI6MTcwMTQyMjcxOSwiYXV0aF90aW1lIjoxNzAxNDIwMjkxLCJqdGkiOiJlMDMwODcxMS0yZTdmLTQ5ODktOTMzNy1kMzVkNTM0YjE2ZGUiLCJpc3MiOiJodHRwczovL3Nzby5kZXYudG9sa2V2YXJhdi5la2kuZWUvcmVhbG1zL3RvbGtldmFyYXYtZGV2IiwiYXVkIjoiYWNjb3VudCIsInN1YiI6IjY5MmQwYzdlLTQzNTgtNDY5MS1hNTgyLTcwYjY3Zjk3M2JlMiIsInR5cCI6IkJlYXJlciIsImF6cCI6IndlYiIsIm5vbmNlIjoiODA3YjQ2ZjgtNTYzZC00YTM3LWEwNDItYzM0ZGM5N2RhNTQyIiwic2Vzc2lvbl9zdGF0ZSI6Ijk1YTFkZTBlLTA4ZmEtNGYwZi1hNmE4LTZjNzBjNmM2YzA5MSIsImFjciI6IjAiLCJhbGxvd2VkLW9yaWdpbnMiOlsiKiJdLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsib2ZmbGluZV9hY2Nlc3MiLCJkZWZhdWx0LXJvbGVzLXRvbGtldmFyYXYtZGV2IiwidW1hX2F1dGhvcml6YXRpb24iXX0sInJlc291cmNlX2FjY2VzcyI6eyJhY2NvdW50Ijp7InJvbGVzIjpbIm1hbmFnZS1hY2NvdW50IiwibWFuYWdlLWFjY291bnQtbGlua3MiLCJ2aWV3LXByb2ZpbGUiXX19LCJzY29wZSI6Im9wZW5pZCBlbWFpbCBwcm9maWxlIiwic2lkIjoiOTVhMWRlMGUtMDhmYS00ZjBmLWE2YTgtNmM3MGM2YzZjMDkxIiwidG9sa2V2YXJhdiI6eyJwZXJzb25hbElkZW50aWZpY2F0aW9uQ29kZSI6IjM5NjA4MTcyNzY1IiwidXNlcklkIjoiOWExZjUyYTAtNWEwOC00ZWNlLWJiMTAtMWVhNDQ1N2U1MDA3IiwiaW5zdGl0dXRpb25Vc2VySWQiOiI5YTFmNTJhMC01YjE3LTQ5N2YtOTkxOC03NTgxMjczODE4MDkiLCJmb3JlbmFtZSI6Ik1hcmt1cyIsInN1cm5hbWUiOiJQZXRlcnNvbiIsInNlbGVjdGVkSW5zdGl0dXRpb24iOnsiaWQiOiI5YTFmNTI5Yy1jYTgwLTQxZWYtOGVjOS0zM2Q1ZDM5YmJiYjEiLCJuYW1lIjoiT2ZpY2luYSBkZSBJZGlvbWFzIGRlIE1hcnF1aXoifSwicHJpdmlsZWdlcyI6WyJBRERfUk9MRSIsIlZJRVdfUk9MRSIsIkVESVRfUk9MRSIsIkRFTEVURV9ST0xFIiwiQUREX1VTRVIiLCJFRElUX1VTRVIiLCJWSUVXX1VTRVIiLCJFWFBPUlRfVVNFUiIsIkFDVElWQVRFX1VTRVIiLCJERUFDVElWQVRFX1VTRVIiLCJBUkNISVZFX1VTRVIiLCJFRElUX1VTRVJfV09SS1RJTUUiLCJFRElUX1VTRVJfVkFDQVRJT04iLCJBRERfVEFHIiwiRURJVF9UQUciLCJERUxFVEVfVEFHIiwiQUREX0RFUEFSVE1FTlQiLCJFRElUX0RFUEFSVE1FTlQiLCJERUxFVEVfREVQQVJUTUVOVCIsIlZJRVdfVkVORE9SX0RCIiwiRURJVF9WRU5ET1JfREIiLCJWSUVXX0dFTkVSQUxfUFJJQ0VMSVNUIiwiVklFV19WRU5ET1JfVEFTSyIsIkVESVRfSU5TVElUVVRJT04iLCJFRElUX0lOU1RJVFVUSU9OX1dPUktUSU1FIiwiQ1JFQVRFX1BST0pFQ1QiLCJNQU5BR0VfUFJPSkVDVCIsIlJFQ0VJVkVfQU5EX01BTkFHRV9QUk9KRUNUIiwiVklFV19QRVJTT05BTF9QUk9KRUNUIiwiVklFV19JTlNUSVRVVElPTl9QUk9KRUNUX0xJU1QiLCJWSUVXX0lOU1RJVFVUSU9OX1BST0pFQ1RfREVUQUlMIiwiQ0hBTkdFX0NMSUVOVCIsIklNUE9SVF9UTSIsIkNSRUFURV9UTSIsIkVYUE9SVF9UTSIsIkVESVRfVE1fTUVUQURBVEEiLCJFRElUX1RNIiwiREVMRVRFX1RNIiwiVklFV19UTSIsIkVESVRfSU5TVElUVVRJT05fUFJJQ0VfUkFURSIsIlZJRVdfSU5TVElUVVRJT05fUFJJQ0VfUkFURSIsIlZJRVdfQVVESVRfTE9HIiwiRVhQT1JUX0FVRElUX0xPRyJdfSwiZW1haWxfdmVyaWZpZWQiOmZhbHNlLCJuYW1lIjoiZWUzOTYwODE3Mjc2NS1maXJzdC1uYW1lIGVlMzk2MDgxNzI3NjUtbGFzdC1uYW1lIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiZWUzOTYwODE3Mjc2NSIsImdpdmVuX25hbWUiOiJlZTM5NjA4MTcyNzY1LWZpcnN0LW5hbWUiLCJmYW1pbHlfbmFtZSI6ImVlMzk2MDgxNzI3NjUtbGFzdC1uYW1lIn0.3vHFVvBnW6yynUgPSbsXXV0IokNjH41uq6G3ZVsQSzMNxlXBGJ4gANAwknUb3jWfFCTfkO5vvPWgNEclBUiD8tbt_ie8-Je4vOVZg2nqnAUVsf2MLzGN-Y89W7kXS8Xn_Z0fVaAK363AT9rEGPwKZdN408K_cMryKTI7No84YsymK_bRS3swpqGnWs5dF3-CC621Hk4UqJt0FUWQPO5Hq-fXwuFeHe4Z3dQ0JWOeK2tftXUaexi7fHomzEJeDtA4ysHC7oK_KosA3xJICRYQyyKoprLpo3phpu22RMAjLrhZnk4HW4ZO0UmLHnMhnA2eA0jJZxaktLbSLuavXwz1dA' \
  -H 'origin: http://localhost:3000' \
  -H 'referer: http://localhost:3000/' \
  -H 'sec-ch-ua: "Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36' \
  --compressed`
  
  3. Currelty I can't add any new skill to language pair. I get BE error: message: "Price already exists (and 3 more errors)"
`curl 'https://api.dev.tolkevarav.eki.ee/translation-order/api/prices/bulk' \
  -H 'authority: api.dev.tolkevarav.eki.ee' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJBRzRYOXR2SVhubkoxWUhiYnJHZmdYdnNYNEVVU2xGalk0RkZSekY5MjlRIn0.eyJleHAiOjE3MDE0MjQ2NjMsImlhdCI6MTcwMTQyMzc2MywiYXV0aF90aW1lIjoxNzAxNDIwMjkxLCJqdGkiOiI0OWQ5MGY3OS0xZjM2LTQ4NjktODRkOS0xZjY3MDc2YTlmYmIiLCJpc3MiOiJodHRwczovL3Nzby5kZXYudG9sa2V2YXJhdi5la2kuZWUvcmVhbG1zL3RvbGtldmFyYXYtZGV2IiwiYXVkIjoiYWNjb3VudCIsInN1YiI6IjY5MmQwYzdlLTQzNTgtNDY5MS1hNTgyLTcwYjY3Zjk3M2JlMiIsInR5cCI6IkJlYXJlciIsImF6cCI6IndlYiIsIm5vbmNlIjoiZTY1MTc5ODQtZGNjYi00NmMzLThiOGYtZjRkZjczZDVkMTc2Iiwic2Vzc2lvbl9zdGF0ZSI6Ijk1YTFkZTBlLTA4ZmEtNGYwZi1hNmE4LTZjNzBjNmM2YzA5MSIsImFjciI6IjAiLCJhbGxvd2VkLW9yaWdpbnMiOlsiKiJdLCJyZWFsbV9hY2Nlc3MiOnsicm9sZXMiOlsib2ZmbGluZV9hY2Nlc3MiLCJkZWZhdWx0LXJvbGVzLXRvbGtldmFyYXYtZGV2IiwidW1hX2F1dGhvcml6YXRpb24iXX0sInJlc291cmNlX2FjY2VzcyI6eyJhY2NvdW50Ijp7InJvbGVzIjpbIm1hbmFnZS1hY2NvdW50IiwibWFuYWdlLWFjY291bnQtbGlua3MiLCJ2aWV3LXByb2ZpbGUiXX19LCJzY29wZSI6Im9wZW5pZCBlbWFpbCBwcm9maWxlIiwic2lkIjoiOTVhMWRlMGUtMDhmYS00ZjBmLWE2YTgtNmM3MGM2YzZjMDkxIiwidG9sa2V2YXJhdiI6eyJwZXJzb25hbElkZW50aWZpY2F0aW9uQ29kZSI6IjM5NjA4MTcyNzY1IiwidXNlcklkIjoiOWExZjUyYTAtNWEwOC00ZWNlLWJiMTAtMWVhNDQ1N2U1MDA3IiwiaW5zdGl0dXRpb25Vc2VySWQiOiI5YTFmNTJhMC01YjE3LTQ5N2YtOTkxOC03NTgxMjczODE4MDkiLCJmb3JlbmFtZSI6Ik1hcmt1cyIsInN1cm5hbWUiOiJQZXRlcnNvbiIsInNlbGVjdGVkSW5zdGl0dXRpb24iOnsiaWQiOiI5YTFmNTI5Yy1jYTgwLTQxZWYtOGVjOS0zM2Q1ZDM5YmJiYjEiLCJuYW1lIjoiT2ZpY2luYSBkZSBJZGlvbWFzIGRlIE1hcnF1aXoifSwicHJpdmlsZWdlcyI6WyJBRERfUk9MRSIsIlZJRVdfUk9MRSIsIkVESVRfUk9MRSIsIkRFTEVURV9ST0xFIiwiQUREX1VTRVIiLCJFRElUX1VTRVIiLCJWSUVXX1VTRVIiLCJFWFBPUlRfVVNFUiIsIkFDVElWQVRFX1VTRVIiLCJERUFDVElWQVRFX1VTRVIiLCJBUkNISVZFX1VTRVIiLCJFRElUX1VTRVJfV09SS1RJTUUiLCJFRElUX1VTRVJfVkFDQVRJT04iLCJBRERfVEFHIiwiRURJVF9UQUciLCJERUxFVEVfVEFHIiwiQUREX0RFUEFSVE1FTlQiLCJFRElUX0RFUEFSVE1FTlQiLCJERUxFVEVfREVQQVJUTUVOVCIsIlZJRVdfVkVORE9SX0RCIiwiRURJVF9WRU5ET1JfREIiLCJWSUVXX0dFTkVSQUxfUFJJQ0VMSVNUIiwiVklFV19WRU5ET1JfVEFTSyIsIkVESVRfSU5TVElUVVRJT04iLCJFRElUX0lOU1RJVFVUSU9OX1dPUktUSU1FIiwiQ1JFQVRFX1BST0pFQ1QiLCJNQU5BR0VfUFJPSkVDVCIsIlJFQ0VJVkVfQU5EX01BTkFHRV9QUk9KRUNUIiwiVklFV19QRVJTT05BTF9QUk9KRUNUIiwiVklFV19JTlNUSVRVVElPTl9QUk9KRUNUX0xJU1QiLCJWSUVXX0lOU1RJVFVUSU9OX1BST0pFQ1RfREVUQUlMIiwiQ0hBTkdFX0NMSUVOVCIsIklNUE9SVF9UTSIsIkNSRUFURV9UTSIsIkVYUE9SVF9UTSIsIkVESVRfVE1fTUVUQURBVEEiLCJFRElUX1RNIiwiREVMRVRFX1RNIiwiVklFV19UTSIsIkVESVRfSU5TVElUVVRJT05fUFJJQ0VfUkFURSIsIlZJRVdfSU5TVElUVVRJT05fUFJJQ0VfUkFURSIsIlZJRVdfQVVESVRfTE9HIiwiRVhQT1JUX0FVRElUX0xPRyJdfSwiZW1haWxfdmVyaWZpZWQiOmZhbHNlLCJuYW1lIjoiZWUzOTYwODE3Mjc2NS1maXJzdC1uYW1lIGVlMzk2MDgxNzI3NjUtbGFzdC1uYW1lIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiZWUzOTYwODE3Mjc2NSIsImdpdmVuX25hbWUiOiJlZTM5NjA4MTcyNzY1LWZpcnN0LW5hbWUiLCJmYW1pbHlfbmFtZSI6ImVlMzk2MDgxNzI3NjUtbGFzdC1uYW1lIn0.4x_DLutspLZxb0-vjP7Zv4Q3lbH4kAzCmIHXQSt5SQzuQf_4JEqBKrmgYDdnb28BwPfxqs0uFrNWeGVzMlx8BTAPLfJjDvSzW_cUQ3fayyTGJO6M58IIo1MXCAJdeMyNus8HwtZ1P-tAhokdBFHKp7ACI1l5FZ41orLb4QmC_iwWMtHK2mGh_Ai8koTIXCzmX07sh4j6DKsvnfF5Hqmd8BqrDaLOQmekSvUoZmJTVwIHUhrwdgf79WawDfVPvaJ-IIt4tF2Iomj-qnUuH8GjBUJPTUrvWEJuPFC6x2wscha6KPvQrrqd8nfW51YUqegwN6O-YhL_U938Z79VsiyUgw' \
  -H 'content-type: application/json' \
  -H 'origin: http://localhost:3000' \
  -H 'referer: http://localhost:3000/' \
  -H 'sec-ch-ua: "Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36' \
  --data-raw '{"data":[{"skill_id":"9a722330-24c1-4c77-8b29-74ea1b1bb41c","vendor_id":"9a8e596a-588c-4efd-ab0e-16c57502f4f3","src_lang_classifier_value_id":"85858551-9da9-44a4-9575-94d47205f2a9","dst_lang_classifier_value_id":"ef5dfd78-3495-45c6-ad7d-15fabb1c3b6e","character_fee":"112","hour_fee":"0","minimal_fee":"0","minute_fee":"0","page_fee":"0","word_fee":"0"}]}' \
  --compressed`

_Originally posted by @VaabelMaret in https://github.com/keeleinstituut/tv-tolkevarav/issues/469#issuecomment-1835751649_

**Comment by MariusJulius** [comment]
1.Low for now.
2. remove pagination 100 option
3. @KaarelKa check. the third one. corner case - if you don't edit but add again same pairs with same skill (as it exists).

**Comment by MariusJulius** [comment]
1. Low for now - not in use anywhere. Need to be considered in the future in a separate ticket.
2. Removed.
3. Corner case issue (8h)
